src/circuit.rs

```diff
@@ -301,6 +301,7 @@ where
     }
 
     /// Return the list of public inputs generated by the gadget
+    #[allow(clippy::wrong_self_convention)]
```

Comment:
Oh, my bad there. Maybe rename to `to_public_inputs`?

Reply:
This is a draft PR, I was planning to go over all the `allowing` part – at least for the things introduce in the branch I'm merging  – since there are other issue like that around the code base, and we could have an issue to tackle them all together.